modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -822,17 +924,22 @@ sub apply_tweaks {
 
                 foreach my $rc (@$resource_classes) {
                     my $rc_name = $rc->name;
+                    $tweakStructure->{Object}->{Id} = $rc->dbID;
```

Comment:
`$tweakStructure` needs to be reset every loop otherwise the `Tweaks` list will contain references to the same object